main.go

```diff
@@ -3,147 +3,118 @@ package main
 	"net/http"
 	"net/url"
-	"os"
+	aws "oncall-notify/pkg/config/aws"
+	slack "oncall-notify/pkg/config/slack"
+	"oncall-notify/pkg/env"
 	"strings"
-	"time"
 
 	"github.com/aws/aws-lambda-go/lambda"
-	"github.com/aws/aws-sdk-go-v2/aws"
-	"github.com/aws/aws-sdk-go-v2/config"
-	"github.com/aws/aws-sdk-go-v2/service/ssmcontacts"
+	jsoniter "github.com/json-iterator/go"
 )
 
-const weekDuration = time.Hour * 24 * 7
+var config *Config
 
-var (
-	client           *ssmcontacts.Client
-	slackWebhookUrl  string
-	slackApiUrl      string
-	slackApiToken    string
-	slackSubTeamID   string
-	slackSubTeamName string
-	slackUserIDs     map[string]string
-)
+type Config struct {
+	AwsConfig   *aws.AwsConfig
+	SlackConfig *slack.SlackConfig
+}
 
-func getEnvOrDefault(key, defaultValue string) string {
-	value := os.Getenv(key)
-	if value == "" {
-		return defaultValue
+func NewConfig(awsConfig *aws.AwsConfig, slackConfig *slack.SlackConfig) *Config {
+	return &Config{
+		AwsConfig:   awsConfig,
+		SlackConfig: slackConfig,
 	}
-	return value
 }
 
-func init() {
-	awsRegion := getEnvOrDefault("AWS_REGION", "eu-west-1")
-	slackApiUrl = getEnvOrDefault("SLACK_API_URL", "https://slack.com/api/usergroups.users.update")
-	slackWebhookUrl = getEnvOrDefault("SLACK_WEBHOOK_URL", "")
-	slackApiToken = getEnvOrDefault("SLACK_API_TOKEN", "")
-	slackSubTeamID = getEnvOrDefault("SLACK_SUBTEAM_ID", "")
-	slackSubTeamName = getEnvOrDefault("SLACK_SUBTEAM_NAME", "support")
+func main() {
+	awsRegion := env.DefaultEnv("AWS_REGION", "eu-west-1")
+	slackApiUrl := env.DefaultEnv("SLACK_API_URL", "https://slack.com/api/usergroups.users.update")
+	slackWebhookUrl := env.DefaultEnv("SLACK_WEBHOOK_URL", "")
+	slackApiToken := env.DefaultEnv("SLACK_API_TOKEN", "")
+	slackSubTeamID := env.DefaultEnv("SLACK_SUBTEAM_ID", "")
+	slackSubTeamName := env.DefaultEnv("SLACK_SUBTEAM_NAME", "support")
+	slackUserIDs := slack.GetUserID()
 
 	if slackApiToken == "" {
-		log.Fatalf("error: Slack API token is not set")
+		fmt.Println("Error: Slack API token is not set")
+		return
 	}
 
-	// Load all environment variables starting with LHV_ into a map
-	slackUserIDs = make(map[string]string)
-
-	for _, e := range os.Environ() {
-		pair := strings.SplitN(e, "=", 2)
-		if strings.HasPrefix(pair[0], "SSM_") {
-			slackUserIDs[strings.ToLower(pair[0])] = pair[1]
-		}
-	}
-
-	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithRegion(awsRegion))
-	if err != nil {
-		log.Fatalf("error loading AWS configuration: %v\n", err)
-	}
-
-	client = ssmcontacts.NewFromConfig(cfg, func(options *ssmcontacts.Options) {
-		options.Region = awsRegion
-	})
-}
-
-func loadAndPrintAllRotationShifts(ctx context.Context) ([]string, error) {
-	outputRotations, err := client.ListRotations(ctx, &ssmcontacts.ListRotationsInput{})
+	awsConfig, err := aws.NewConfig(awsRegion)
 	if err != nil {
-		return nil, fmt.Errorf("error loading rotations: %w", err)
+		fmt.Printf("Error creating AWS config: %v\n", err)
+		return
```

Comment:
```suggestion
		log.Fatalf("Error creating AWS config: %v\n", err)
```
Lambda should fail if it cannot get AWS Config